core/frontend/src/tile/map/ImageryTileTree.ts

```diff
@@ -142,6 +151,35 @@ export class ImageryMapTile extends RealityTile {
   }
 }
 
+/** Object that holds various state values for a MapTileTree
+ * @internal */
+export class MapTileTreeState {
+  private _scaleRangeVis: MapTileTreeScaleRangeVisibility;
+  public getScaleRangeVisibility() {return this._scaleRangeVis;}
+
+  constructor() {
+    this._scaleRangeVis = MapTileTreeScaleRangeVisibility.Unknown;
+  }
+
+  public clone() {
+    const clone = new MapTileTreeState();
+    clone._scaleRangeVis = this._scaleRangeVis;

```

Comment:
Does this intentionally not make a deep copy of the scale range visibility?

Reply:
MapTileTreeScaleRangeVisibility is an enum, by using the assign operator it will make a copy of it, no?

Reply:
Yeah. For some reason I thought it was an object.

Reply:
Seems like a lot to invent a whole class and allocate instances of it to hold a single enum value, especially one with a new like `MapTileTreeState` which suggests it contains something more substantial.

Reply:
In my book there is no harm in having small classes.  Is it a naming issue or a design issue? What do you suggest?

Reply:
You've introduced one new concept - a set of possible "scale range visibility" values. To represent this concept you've introduced two new types - an enum, and a "state" object containing a value of that enum type.
Each type you introduce increases the complexity of your solution.
The "state" object adds nothing but overhead, both in terms of performance (heap-allocating multiple garbage-collected objects just for the sake of storing/comparing integer enum values) and code clarity.

Do you have compelling reason to believe that in the future, more information will be added to MapTileTreeState?

Reply:
I don't have short term needs for addition state value, I'm just supposing might be added in the future.  If you prefer I can change `MapTileTree_imageryTreeState` to be defined as `Map<Id64String, MapTileTreeScaleRangeVisibility>`.

Doing so, will force me to move related logic (i.e. `MapTileTreeState.setScaleRangeVisibility`) somewhere else (instead of keeping it on a dedicated class), making it even harder to read, but you decide.  Also if we want to add a new ImageryTileTree state in the future, we will have to create yet another Map object, thus increasing memory consumption your are trying to avoid.

Reply:
Do what you want I guess, but if you keep this `MapTileTreeState` concept please make it `@internal` and document it.

Reply:
I renamed it for ImageryTileTreeState for clarity.  Its already internal and documented.